Issue #1118 Update docs for registry container

mikehaertl opened the pull request:
@solidnerd I was very courageous with my changes - I hope you don't mind.

What I did:

* Add a section "Configuration" after "Installation" and move the config settings there
* Remove the full `docker-compose.yml` example as it hides the really relevant parts for the registry. We already explain the Gitlab setup in the main README.
* Remove the part about official certificats. As I see it, they are not needed. And if someone wants to use them, they will know what to do.
* Only list the relevant changes in docker-compose.yml and for the Nginx registry setup

If you feel that I changed it too much, let me know and I can change it back.

solidnerd commented on docs/container_registry.md:
````diff
@@ -63,134 +185,6 @@ gitlab:
-    - REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE=/certs/registry-auth.crt
-    - REGISTRY_STORAGE_DELETE_ENABLED=true
-```
-> **Important Notice**
````
I think we should keep this information

mikehaertl replied:
@solidnerd Added it back in.

solidnerd commented on docs/container_registry.md:
```diff
@@ -16,7 +16,129 @@ Since `8.8.0` GitLab introduces container registry. GitLab is helping to authent
+## Setup with Nginx as Reverse Proxy
+
+We assume that you already have Nginx installed on your host system and that
+you use a reverse proxy configuration to connect to your Gitlab container.
```
Replace Gitlab with GitLab

mikehaertl commented:
@solidnerd I've fixed Gitlab to GitLab everywhere and also rebased the changes to master to fix a merge conflict due to 10.1.2 upgrade.

Let me know if there's more that should be changed.

solidnerd commented:
@mikehaertl LGTM !
